include/shadesmar/robust_lock.h

```diff
@@ -66,6 +68,14 @@ void RobustLock::lock() {
   exclusive_owner = getpid();
 }
 
+bool RobustLock::try_lock() {
+  bool acquired = mutex_.try_lock();
+  if(acquired) {
+    exclusive_owner = getpid();
+  }
+  return acquired;
+}
```

Comment:
It's not robust. 

1. If `acquire` is true, everything is good. 
2. If `acquire` is false, (`try_lock()` failed). There are two possible scenarios:
a. Some other live process has the lock, so we exit. This is fine as well.
b. Some other process held a write or read lock, and then died. We'll need to do a check similar to `lock()` to figure out if the last process(es) that held the lock are still alive.